lib/_internal/file_picker_web.dart

```diff
@@ -184,6 +184,7 @@ class FilePickerWeb extends FilePicker {
     }
 
     final List<PlatformFile>? files = await filesCompleter.future;
+    filesCompleter = null;
```

Comment:
Should use `try/catch/finally` for the function, to avoid the case of an error occurring before we can clean up `filesCompleter`

```dart
try {
   ......
} catch (e) {
   filesCompleter.completeError(e);
    rethrow;
} finally {
   filesCompleter = null;
}
```